packages/opentelemetry-core/src/trace/instrumentation/BasePlugin.ts

```diff
@@ -14,20 +14,24 @@
  * limitations under the License.
  */
 
-import { Tracer, Plugin } from '@opentelemetry/types';
+import { Tracer, Plugin, Logger } from '@opentelemetry/types';
+import { NoopLogger } from '../../common/NoopLogger';
 
 /** This class represent the base to patch plugin. */
 export abstract class BasePlugin<T> implements Plugin<T> {
   protected _moduleExports!: T;
   protected _tracer!: Tracer;
+  protected _logger!: Logger;
 
   enable(
     moduleExports: T,
     tracer: Tracer,
+    logger?: Logger,
```

Comment:
I think it would be a problem if you give no logger but a config to the function ? Especially from javascript where you don't have the type ?
The logger will be an config object and the logging will throw error

Reply:
I believe it should be the case yeah